### PR TITLE
[SPARK-41127][CONNECT][PYTHON] Implement DataFrame.CreateGlobalView in Python client

### PR DESCRIPTION
**Pull request by amaliujia**
### What changes were proposed in this pull request?

This PR adds `CreateGlobalTempView` and `CreateOrReplaceGlobalTempView` to Python DataFrame API.

Meanwhile, this PR extends `LogicalPlan` to let it have the ability to deal with `Command`.

### Why are the changes needed?

Improve API coverage.

### Does this PR introduce _any_ user-facing change?

NO

### How was this patch tested?

UT

### CODE REVIEW

File: python/pyspark/sql/tests/connect/test_connect_basic.py [hunk 1]
```diff
@@ -207,6 +208,18 @@ def test_range(self):
...
+    def test_create_global_temp_view(self):
+        # SPARK-41127: test global temp view creation.
+        with self.tempView("view_1"):
+            self.connect.sql("SELECT 1 AS X LIMIT 0").createGlobalTempView("view_1")
+            self.connect.sql("SELECT 2 AS X LIMIT 1").createOrReplaceGlobalTempView("view_1")
+            self.assertTrue(self.spark.catalog.tableExists("global_temp.view_1"))
+
+            # Test when creating a view which is alreayd exists but
+            self.assertTrue(self.spark.catalog.tableExists("global_temp.view_1"))
+            with self.assertRaises(_MultiThreadedRendezvous):
```
Comment (amaliujia) [inline python/pyspark/sql/tests/connect/test_connect_basic.py:220]: cc @zhengruifeng @HyukjinKwon 

I found this is what we have now when there is an exception happened during a RPC call.

I guess we need to improve it.

Comment (amaliujia) [reply python/pyspark/sql/tests/connect/test_connect_basic.py:220]: `_MultiThreadedRendezvous` is raised from gRPC channel when gRPC finds response state code is not `OK` (thus some error has happened).

Comment (grundprinzip) [reply python/pyspark/sql/tests/connect/test_connect_basic.py:220]: Yes, right now, the client does not do any error handling, the exception simply bubbles up.

Comment (grundprinzip) [reply python/pyspark/sql/tests/connect/test_connect_basic.py:220]: you can just check for `grpc.RpcError`

Comment (amaliujia) [reply python/pyspark/sql/tests/connect/test_connect_basic.py:220]: @grundprinzip yes! It is much better to use `grpc.RpcError` for readability.

Comment (amaliujia) [reply python/pyspark/sql/tests/connect/test_connect_basic.py:220]: Eventually we should also use error classes in client side (based on the grpc error to map to error classes by a way). Right now we only have this.

File: python/pyspark/sql/connect/dataframe.py [hunk 2]
```diff
@@ -633,6 +633,48 @@ def explain(self) -> str:
...
+        .. versionadded:: 3.4.0
+
+        Parameters
+        ----------
+        name : str
+            Name of the view.
+
+        Returns
+        -------
+        None
```
Comment (HyukjinKwon) [inline python/pyspark/sql/connect/dataframe.py:650]:
```suggestion

        Returns
        -------
        None
```

I guess we won't need this for `None`.

File: python/pyspark/sql/connect/dataframe.py [hunk 3]
```diff
@@ -633,6 +633,48 @@ def explain(self) -> str:
...
```
Comment (HyukjinKwon) [inline python/pyspark/sql/connect/dataframe.py:671]:
```suggestion
```

**Review by HyukjinKwon — APPROVED** [review]

**Review by zhengruifeng — APPROVED** [review]
lgtm

**Comment by HyukjinKwon** [comment]
Merged to master.

**Pull request closed without merging**